pkg/config/vpn-client.go

```diff
@@ -5,40 +5,47 @@
 package config
 
 import (
+	"fmt"
+	"slices"
 	"strconv"
 	"strings"
 	"time"
 
 	"github.com/caarlos0/env/v10"
+
+	"github.com/gardener/vpn2/pkg/constants"
 	"github.com/gardener/vpn2/pkg/network"
 )
 
 type VPNClient struct {
 	TCP struct {
-		KeepAliveTime     int64 `env:"KEEPALIVE_TIME" envDefault:"7200"`
-		KeepAliveInterval int64 `env:"KEEPALIVE_INTVL" envDefault:"75"`
-		KeepAliveProbes   int64 `env:"KEEPALIVE_PROBES" envDefault:"9"`
+		KeepAliveTime     uint64 `env:"KEEPALIVE_TIME" envDefault:"7200"`
+		KeepAliveInterval uint64 `env:"KEEPALIVE_INTVL" envDefault:"75"`
+		KeepAliveProbes   uint64 `env:"KEEPALIVE_PROBES" envDefault:"9"`
 	} `envPrefix:"TCP_"`
-	IPFamilies        string       `env:"IP_FAMILIES" envDefault:"IPv4"`
-	Endpoint          string       `env:"ENDPOINT"`
-	OpenVPNPort       int          `env:"OPENVPN_PORT" envDefault:"8132"`
-	VPNNetwork        network.CIDR `env:"VPN_NETWORK"`
-	SeedPodNetwork    network.CIDR `env:"SEED_POD_NETWORK"`
-	IsShootClient     bool         `env:"IS_SHOOT_CLIENT"`
-	PodName           string       `env:"POD_NAME"`
-	Namespace         string       `env:"NAMESPACE"`
-	VPNServerIndex    string       `env:"VPN_SERVER_INDEX"`
-	VPNClientIndex    int
-	IsHA              bool          `env:"IS_HA"`
-	ReversedVPNHeader string        `env:"REVERSED_VPN_HEADER" envDefault:"invalid-host"`
-	HAVPNClients      int           `env:"HA_VPN_CLIENTS"`
-	HAVPNServers      int           `env:"HA_VPN_SERVERS"`
-	PodLabelSelector  string        `env:"POD_LABEL_SELECTOR" envDefault:"app=kubernetes,role=apiserver"`
-	WaitTime          time.Duration `env:"WAIT_TIME" envDefault:"2s"`
+	IPFamilies           []string       `env:"IP_FAMILIES" envDefault:"IPv4"`
+	Endpoint             string         `env:"ENDPOINT"`
+	OpenVPNPort          uint           `env:"OPENVPN_PORT" envDefault:"8132"`
+	VPNNetwork           network.CIDR   `env:"VPN_NETWORK"`
+	SeedPodNetworkV4     network.CIDR   `env:"SEED_POD_NETWORK_V4"`
+	ShootServiceNetworks []network.CIDR `env:"SHOOT_SERVICE_NETWORKS"`
+	ShootPodNetworks     []network.CIDR `env:"SHOOT_POD_NETWORKS"`
+	ShootNodeNetworks    []network.CIDR `env:"SHOOT_NODE_NETWORKS"`
```

Comment:
How does the `env` package work with slices? Do we need to specify the environment variable multiple times or is there some magic with splitting strings?

Reply:
yes, there is some magic going on. if the string is comma-delimited, it creates an array for each delimited element. so `a,b,c` => `[]string{"a","b","c"}`

have been doing the same already [on the seed side](https://github.com/gardener/vpn2/pull/137/files#diff-90491813c7a3007a8781e4d811f30f169838371f25e8391e3183b5b101e374c4R17) even [before this PR](https://github.com/gardener/vpn2/blob/master/pkg/config/vpn-server.go#L14), so it does work just fine.